core/runtime/jsruntime.rs

```diff
@@ -608,8 +607,7 @@ impl JsRuntime {
     let waker = op_state.waker.clone();
     let op_state = Rc::new(RefCell::new(op_state));
     let state_rc = Rc::new(JsRuntimeState {
-      source_map_getter: options.source_map_getter.map(Rc::new),
-      source_map_cache: Default::default(),
+      source_mapper: RefCell::new(SourceMapper::new(options.source_map_getter)),
```

Comment:
I'm a bit confused here - now we always create `SourceMapper` even if `options.source_map_getter` is not provided. Shouldn't we make `SourceMapper` an `Option` here?

Reply:
There are two reasons I think it is better to not have it as an `Option`
1. When support for extension's source map is present, it will be always present, unless there are no extensions with typescript.
2. It encapsulates the behaviour of when no source map is present, which allows to avoid using `if Some(source_mapper) = ... { ... } else { Unchanged }` at all use sites.

Reply:
Okay, that's reasonable.